src/cozmo/behavior.py

```diff
@@ -68,29 +88,85 @@ def __init__(self, robot, behavior_type, is_active=False, **kw):
         super().__init__(**kw)
         self.robot = robot
         self.type = behavior_type
-        self._is_active = is_active
+        self._state = BEHAVIOR_IDLE
         if is_active:
-            self.dispatch_event(EvtBehaviorStarted, behavior=self, behavior_type_name=self.type.name)
+            self._state = BEHAVIOR_REQUESTED
+            self.dispatch_event(EvtBehaviorRequested, behavior=self, behavior_type_name=self.type.name)
 
     def __repr__(self):
         return '<%s type="%s">' % (self.__class__.__name__, self.type.name)
 
+    def _on_engine_started(self):
+        if self._state != BEHAVIOR_REQUESTED:
+            # has not been requested (is an unrelated behavior transition)
+            if self.is_running:
+                logger.warning("Behavior '%s' reported started when already running!?")
```

Comment:
Would be good to remove '!?" from the warning and instead express that this is "unexpected" or something.